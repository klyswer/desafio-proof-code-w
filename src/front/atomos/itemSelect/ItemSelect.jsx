import React from 'react';
import PropTypes from 'prop-types';
import { agregarformatoPesos } from '../../../utils/FormatUtil';
import './style.css';

const ItemSelect = ({id,marca,src,nombre,cantidad,precio,handleEliminar}) => {
  return (
    <div className="item_default_select">
      <div className="contenedor_item_select">
        <div className="icon_product">
          <div className="walmart_cantidad_item">{cantidad}</div>
          <img src={src!==''&& `https://${src}`} alt={nombre} />
        </div>
        <div className="descript_product">
          <div className="">
            {nombre}  
          </div>       
          <div className="">
            {cantidad}
          </div>        
        </div>

        <div className="precio_product">
          <div>
            <strong>{agregarformatoPesos(precio)}</strong>
          </div>
          <button type="button" name={id} id={marca} onClick={handleEliminar}>Eliminar</button>
        </div>
        
      </div>
    </div>
  );
}

ItemSelect.defaultProps = {
  src:'',
  nombre:'',
  cantidad:1,
  precio:0,
  marca:''
}

ItemSelect.propTypes = {
  id:PropTypes.number.isRequired,
  src:PropTypes.string,
  nombre:PropTypes.string,
  cantidad:PropTypes.number,
  precio:PropTypes.number,
  handleEliminar: PropTypes.func.isRequired,
  marca: PropTypes.string,
};

export default ItemSelect;