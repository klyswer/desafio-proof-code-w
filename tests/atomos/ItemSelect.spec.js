import React from 'react';
/* eslint-disable no-undef */
import { shallow } from 'enzyme';
import ItemSelect from '../../src/front/atomos/itemSelect/ItemSelect';

describe('<ItemSelect />', () => {

  const id='kdi293dl';
  const marca="Marca01";
  const src='aire.cl';
  const nombre="Elm 1";
  const precio=1000;
  const handleEliminar=()=>{};

  const wrapper = shallow( <ItemSelect id={id} marca={marca} src={src} nombre={nombre} precio={precio} handleEliminar={handleEliminar} />);

  test('Primer SnapShot: <ItemSelect />', () => {
    expect(wrapper).toMatchSnapshot();
  });
  
});
