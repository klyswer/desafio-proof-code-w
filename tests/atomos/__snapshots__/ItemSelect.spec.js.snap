// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemSelect /> Primer SnapShot: <ItemSelect /> 1`] = `
<div
  className="item_default_select"
>
  <div
    className="contenedor_item_select"
  >
    <div
      className="icon_product"
    >
      <div
        className="walmart_cantidad_item"
      >
        1
      </div>
      <img
        alt="Elm 1"
        src="https://aire.cl"
      />
    </div>
    <div
      className="descript_product"
    >
      <div
        className=""
      >
        <strong>
          Marca01
        </strong>
      </div>
      <div
        className=""
      >
        Elm 1
      </div>
      <div
        className=""
      >
        <strong>
          Cantidad: 1
        </strong>
      </div>
    </div>
    <div
      className="precio_product"
    >
      <div>
        <strong>
          CLPÂ 1,000
        </strong>
      </div>
      <button
        id="Marca01"
        name="kdi293dl"
        onClick={[Function]}
        type="button"
      >
        Eliminar
      </button>
    </div>
  </div>
</div>
`;
